### PR TITLE
Add Centos7 support for ring-install.sh

### PR DESCRIPTION
**Pull request by cazino**
https://37.187.159.67:5443/job/ring-install-test/69/

Please note that centos7 support is available starting RING version 5.

### CODE REVIEW

File: jenkins/ring-install.sh [hunk 1]
```diff
@@ -71,15 +90,22 @@ function add_source {
 }
 
 function add_source_centos {
+    local centos_version=${os_RELEASE:0:1}
     set +x
     sudo sh -c "cat <<-EOF >/etc/yum.repos.d/scality${RING_VERSION}.repo
 [scality-base]
 name=Centos6 - Scality Base
```
Comment (JordanP) [inline jenkins/ring-install.sh]: Change the name of the repo. I suggest "Centos - Scality Base", no version number, no problemo :)

File: utils/jobtool/cli.py [hunk 2]
```diff
@@ -124,21 +124,16 @@ def interactive_connect(user, ip, ssh_key):
     subprocess.call(" ".join(cmd), shell=True)
 
 
-class ManilaTempestJob(object):
-    """ mandatory job_params : JOB_GIT_REVISION
+class Job(object):
```
Comment (JordanP) [inline utils/jobtool/cli.py]: Could you put that on a separated pull request please ?

**Comment by cazino** [comment]
Some enhancements , I am working on it a little bit more.

**Comment by cazino** [comment]
I think it is ready now.
https://37.187.159.67:5443/job/ring-install-test/78/

File: jenkins/ring-install-test/run.sh [hunk 3]
```diff
@@ -8,16 +8,27 @@ export HOST_IP=$(/sbin/ip addr show dev eth0 | sed -nr 's/.*inet ([0-9.]+).*/\1/
...
-install_ringsh
-build_ring
-show_ring_status
-install_sproxyd
-test_sproxyd
-install_sfused
+GetDistro
+
+if [[ ($os_RELEASE =~ ^7) && ($RING_VERSION -lt 5) ]]; then
+    set +e
```
Comment (JordanP) [inline jenkins/ring-install-test/run.sh]: I am not sure you need "+e" here since you wrap the command that could fail in a "if" statement.

File: jenkins/ring-install-test/run.sh [hunk 4]
```diff
@@ -8,16 +8,27 @@ export HOST_IP=$(/sbin/ip addr show dev eth0 | sed -nr 's/.*inet ([0-9.]+).*/\1/
...
+    if initialize; then
+        echo "initiliaze shlould fail in that configuration"
```
Comment (JordanP) [inline jenkins/ring-install-test/run.sh]: Multiple typo here.

File: jenkins/ring-install-test/run.sh (hunk 3 above)
Comment (cazino) [inline jenkins/ring-install-test/run.sh]: Yes you are right.

**Comment by cazino** [comment]
https://37.187.159.67:5443/job/ring-install-test/79/

File: jenkins/ring-install.sh [hunk 5]
```diff
@@ -71,15 +88,22 @@ function add_source {
...
-baseurl=http://${SCAL_PASS}@packages.scality.com/stable_${RING_CODENAME}/centos/6/x86_64/
+name=Centos - Scality Base
+baseurl=http://${SCAL_PASS}@packages.scality.com/stable_${RING_CODENAME}/centos/${centos_version}/x86_64/
 gpgcheck=0
 EOF"
     set -x
-    sudo rpm -Uvh http://mirror.cogentco.com/pub/linux/epel/6/i386/epel-release-6-8.noarch.rpm
+    local epel_repo
+    if [[ $os_RELEASE =~ ^6 ]]; then
+        epel_repo='http://mirror.cogentco.com/pub/linux/epel/6/i386/epel-release-6-8.noarch.rpm'
```
Comment (JordanP) [inline jenkins/ring-install.sh:102]: Is there a reason we use the i386 arch here ?

File: jenkins/ring-install.sh [hunk 6]
```diff
@@ -392,10 +416,10 @@ EOF
     fi
 
     # The following command must be run only once. It touches data on the ring, it does nothing at the connector's side
-    sudo sfused -X -c /etc/sfused.conf
-    sudo /etc/init.d/scality-sfused restart
+    sudo $(which sfused) -X -c /etc/sfused.conf
```
Comment (JordanP) [inline jenkins/ring-install.sh:419]: For my info, why do we have to use `which` here ?

Comment (cazino) [inline jenkins/ring-install.sh:419]: Well, I don't know why, but on Centos7 the 'sfused' bin is not in PATH if you use sudo.

File: jenkins/ring-install.sh (hunk 5 above)
Comment (cazino) [inline jenkins/ring-install.sh:102]: The url is the one of our documentation. :
https://docs.scality.com/display/R50/Requirements+and+Recommendations+for+Installation#RequirementsandRecommendationsforInstallation-DeclaringtheScalityrepository
It is true that we might want to use the x86_64 version, but I think that should be done in a different PR.

File: jenkins/ring-install.sh (hunk 6 above)
Comment (JordanP) [inline jenkins/ring-install.sh:419]: Yeah I remember something along this line now.

**Comment by JordanP** [comment]
LGTM
